arango_api/services/sunburst_service.py

```
@@ -21,163 +29,312 @@ def __init__(self, message, db_error=None):
     except Exception as e:
-        logger.exception("AQL execution failed for full structure load")
+        logger.exception(
+            "AQL execution failed for phenotypes sunburst (parent=%s)", parent_id
+        )
         db_error = None
         if hasattr(e, "response") and hasattr(e.response, "text"):
             db_error = e.response.text
         raise SunburstServiceError(
-            "Failed to fetch full phenotype structure.", db_error=db_error
+            "Failed to fetch phenotype structure.", db_error=db_error
+        )
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)